### PR TITLE
Minimal definition of interfaces to support modes and difficulties.

### PR DESCRIPTION
**Pull request by mcmachado**
@mgbellemare, I created a first minimal pull request for this feature. In this first PR all I'm doing is defining types and the interface I want to make accessible for the end user. If you are ok with it, I'll proceed to implement the functionalities, to update the manual, and to implement the Python interface as well.

### CODE REVIEW

File: doc/examples/Makefile.sharedLibraryWithModes [hunk 1]
```diff
@@ -0,0 +1,31 @@
+USE_SDL := 0
```
Comment (mgbellemare) [inline doc/examples/Makefile.sharedLibraryWithModes]: Are people using these Makefiles?

File: src/ale_interface.cpp [hunk 2]
```diff
@@ -266,6 +266,36 @@ reward_t ALEInterface::act(Action action) {
   return reward;
 }
 
+// Returns the vector of modes available for the current game.
+// This should be called only after the rom is loaded.
+ModeVect ALEInterface::getAvailableModes(){
```
Comment (mgbellemare) [inline src/ale_interface.cpp]: Space between () and { please (also applies to other methods below).

File: doc/examples/Makefile.sharedLibraryWithModes (hunk 1 above)
Comment (mcmachado) [inline doc/examples/Makefile.sharedLibraryWithModes]: I don't know. But I don't see why it hurts to keep them.

**Comment by mcmachado** [comment]
Closed because I'll work on dev.

**Pull request closed without merging**